debezium-core/src/main/java/io/debezium/pipeline/signal/ExternalSignalThread.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.pipeline.signal;
+
+/**
+ * The class responsible for processing of signals delivered to Debezium via a dedicated Kafka topic.
+ * The signal message must have the following structure:
+ * <ul>
+ * <li>{@code id STRING} - the unique identifier of the signal sent, usually UUID, can be used for deduplication</li>
+ * <li>{@code type STRING} - the unique logical name of the code executing the signal</li>
+ * <li>{@code data STRING} - the data in JSON format that are passed to the signal code
+ * </ul> 
+ */
+public interface ExternalSignalThread {
+    void start();
+
+    void seek(long signalOffset);
```

Comment:
is this correct interface to start implementations with? seek method is mostly kafka related, for examlpe file based  ExternalSignal not sure of seek method required.